core/pom.xml

```diff
@@ -313,6 +328,11 @@
             <version>${spark.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.apache.maven</groupId>
+            <artifactId>maven-artifact</artifactId>
```

Comment:
Curious why this dependency is required.

Reply:
Thanks @nartal1 
The dependency has the "ComparableVersion" classes which allows to compare the versions.
I considered using an existing jar because it has the features we need and can recognize mix between alphanumerics.
The dependency is shaded and added to the final jar 

There are other alternatives:

- Implement the comparison in house. I found that unnecessary especially that we will have to handle different variations including "snapshots, CSP versions which can be sometyhing like 3.4.0-cspName..etc"
- Hadoop-Util has that class built-in but I did not want to rely on that in case the jar is not part of the CP (depending on the user's environment)
- there is a gradle module, but this would require adding gradle repository to the pom file

Does that make sense?

Reply:
Thanks for the explanation, @amahussein . Using the jar makes sense instead of the alternatives that were considered.